cfme/tests/cloud_infra_common/test_provisioning.py

```diff
@@ -430,53 +461,58 @@ def test_provision_with_boot_volume(request, instance_args, provider, soft_asser
+            )
+        '''.format(volume))
+
+    @request.addfinalizer
+    def _finish_method():
         with update(method):
-            method.script = dedent('''\
-                $evm.root["miq_provision"].set_option(
-                    :clone_options, {{
-                        :image_ref => nil,
-                        :block_device_mapping_v2 => [{{
-                            :boot_index => 0,
-                            :uuid => "{}",
-                            :device_name => "vda",
-                            :source_type => "volume",
-                            :destination_type => "volume",
-                            :volume_size => 1,
-                            :delete_on_termination => false
-                        }}]
-                    }}
-                )
-            '''.format(volume))
-
-        @request.addfinalizer
-        def _finish_method():
-            with update(method):
-                method.script = """prov = $evm.root["miq_provision"]"""
-
-        instance = appliance.collections.cloud_instances.create(vm_name,
-                                                                provider,
-                                                                form_values=inst_args)
-
-        request_description = 'Provision from [{}] to [{}]'.format(image,
-                                                                   instance.name)
-        provision_request = appliance.collections.requests.instantiate(request_description)
-        try:
-            provision_request.wait_for_request(method='ui')
-        except Exception as e:
-            logger.info(
-                "Provision failed {}: {}".format(e, provision_request.request_state))
-            raise
-        msg = "Provisioning failed with the message {}".format(
-            provision_request.row.last_message.text)
-        assert provision_request.is_succeeded(method='ui'), msg
-        soft_assert(instance.name in provider.mgmt.volume_attachments(volume))
-        soft_assert(provider.mgmt.volume_attachments(volume)[instance.name] == "/dev/vda")
+            method.script = """prov = $evm.root["miq_provision"]"""
+
+    instance = appliance.collections.cloud_instances.create(vm_name,
+                                                            provider,
+                                                            form_values=inst_args)
+
+    @request.addfinalizer
+    def delete_vm_and_wait_for_gone():
         instance.mgmt.delete()  # To make it possible to delete the volume
         wait_for(lambda: not instance.exists_on_provider, num_sec=180, delay=5)
 
+    request_description = 'Provision from [{}] to [{}]'.format(image, instance.name)
+    provision_request = appliance.collections.requests.instantiate(request_description)
+    try:
+        provision_request.wait_for_request(method='ui')
+    except Exception as e:
+        logger.info(
+            "Provision failed {}: {}".format(e, provision_request.request_state))
```

Comment:
optional:
while formatting `logger` string. I would use `%s`

Reply:
@digitronik Why is this? Just curious.

Reply:
Again, this is just some part of already written code that I just had to shuffle around to make tests working and properly clean after.

@john-dupuy I agree what the @digitronik's says because it is shorter than .format(). Logger methods are formatting the messages themselves with the older `"%s"`, so it would look like `logger.info("Provision failed %s: $s", e, provison_request.request_state)`